test/gtest/uct/ib/test_ib.cc

```diff
@@ -574,30 +654,52 @@ UCS_TEST_F(test_uct_ib_sl_utils, sl_selection) {
-        select_sl(ar_enable, 8, 0,
-                  (ar_enable == UCS_YES) ? err_status : UCS_OK);
-
-        /* select SL=8, without OOO SL mask (not detected) */
-        select_sl(ar_enable, 8, m_ooo_sl_mask_not_detected,
-                  ((ar_enable != UCS_TRY) && (ar_enable != UCS_AUTO)) ?
-                  err_status : UCS_OK);
-
-        /* select SL=8, with OOO SL mask { 8 } */
-        select_sl(ar_enable, 8, UCS_BIT(8),
-                  (ar_enable == UCS_NO) ? err_status : UCS_OK);
-
-        /* select SL=8, with OOO SL mask { 0 } */
-        select_sl(ar_enable, 8, UCS_BIT(0),
-                  (ar_enable == UCS_YES) ? err_status : UCS_OK);
-
-        /* select SL=8, with OOO SL mask { 0, 4, 8 } */
-        select_sl(ar_enable, 8, UCS_BIT(0) | UCS_BIT(4) | UCS_BIT(8),
-                  (ar_enable == UCS_NO) ? err_status : UCS_OK);
-
-        /* select SL=8, with OOO SL mask { 0, 8, 11 } */
-        select_sl(ar_enable, 8, UCS_BIT(0) | UCS_BIT(4) | UCS_BIT(11),
-                  (ar_enable == UCS_YES) ? err_status : UCS_OK);
+        for (uint8_t sl = 0; sl < UCT_IB_SL_NUM; ++sl) {
+            /* select SL=<sl>, with empty OOO SL mask */
+            select_sl(ar_enable, sl, 0,
+                      (ar_enable == UCS_YES) ? err_status : UCS_OK);
+
+            /* select SL=<sl>, without OOO SL mask (not detected) */
+            select_sl(ar_enable, sl, m_ooo_sl_mask_not_detected,
+                      ((ar_enable != UCS_TRY) && (ar_enable != UCS_AUTO)) ?
+                      err_status : UCS_OK);
+
+            /* select SL=<sl>, with OOO SL mask { sl } */
+            select_sl(ar_enable, sl, UCS_BIT(sl),
+                      (ar_enable == UCS_NO) ? err_status : UCS_OK);
+
+            /* select SL=<sl>, with OOO SL mask { UCT_IB_SL_NUM - 1 - sl },
+             * i.e. OOO SL mask that doesn't contain <sl> */
+            select_sl(ar_enable, sl, UCS_BIT(UCT_IB_SL_NUM - 1 - sl),
+                      (ar_enable == UCS_YES) ? err_status : UCS_OK);
+
+            /* select SL=<sl>, with OOO SL mask { (sl - 1) % UCT_IB_SL_NUM,
+                                                  sl,
+                                                  (sl + 1) % UCT_IB_SL_NUM } */
+            select_sl(ar_enable, sl,
+                      UCS_BIT((static_cast<int8_t>(sl) - 1) % UCT_IB_SL_NUM) |
+                      UCS_BIT(sl)                                            |
+                      UCS_BIT((sl + 1) % UCT_IB_SL_NUM),
+                      (ar_enable == UCS_NO) ? err_status : UCS_OK);
+
+            /* select SL=<sl>, with OOO SL mask { (sl - 1) % UCT_IB_SL_NUM,
+                                                  UCT_IB_SL_NUM - 1 - sl,
+                                                  (sl + 1) % UCT_IB_SL_NUM },
+             * i.e. OOO SL mask that doesn't contain <sl> */
+            select_sl(ar_enable, sl,
+                      UCS_BIT((static_cast<int8_t>(sl) - 1) % UCT_IB_SL_NUM) |
```

Comment:
what should it do when sl==0?

Reply:
it will `15`

Reply:
picked the simplification of the test from the #6037